File: demo/nested-layouts/nested-layouts.cpp
#include <QtWidgets>

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
    QWidget window;

    QLabel *queryLabel = new QLabel(
        QApplication::translate("nestedlayouts", "Query:"));
    QLineEdit *queryEdit = new QLineEdit();
    QTableView *resultView = new QTableView();

    QHBoxLayout *queryLayout = new QHBoxLayout();
    queryLayout->addWidget(queryLabel);
    queryLayout->addWidget(queryEdit);

    QVBoxLayout *mainLayout = new QVBoxLayout();
    mainLayout->addLayout(queryLayout);
    mainLayout->addWidget(resultView);
    window.setLayout(mainLayout);

    window.setWindowTitle(
        QApplication::translate("nestedlayouts", "Nested layouts"));
    window.show();
    return app.exec();
} 

File: demo/nested-layouts/nested-layouts.ts
import { qt } from "../../src/qt-jsx/qt-export";

// Create application and store in global scope
const app = new qt.QApplication();
(global as any).app = app;

// Create main window and store in global scope
const qwindow = new qt.QWidget();

// Create widgets and store in global scope
const queryLabel = new qt.QLabel("Query:");
const queryEdit = new qt.QLineEdit();
const resultView = new qt.QTableView();
(global as any).widgets = {
    queryLabel,
    queryEdit,
    resultView
};

// Create layouts and store in global scope
const queryLayout = new qt.QHBoxLayout();
queryLayout.addWidget(queryLabel);
queryLayout.addWidget(queryEdit);

const mainLayout = new qt.QVBoxLayout();
mainLayout.addLayout(queryLayout);
mainLayout.addWidget(resultView);
(global as any).layouts = {
    queryLayout,
    mainLayout
};

// Set layout for window
qwindow.setLayout(mainLayout);

// Set window title and show
qwindow.setWindowTitle("Nested layouts");
qwindow.show();

// Start event loop
app.exec(); 